.travis.yml

```diff
@@ -2,10 +2,17 @@ sudo: required
 # trusty required for docker
 # https://github.com/travis-ci/travis-ci/issues/5448
 dist: trusty
+go:
+- 1.7
 services:
 - docker
+install:
+- echo ${GOPATH}
+- mkdir -p ${GOPATH}/k8s.io
+- ln -s `pwd` ${GOPATH}/k8s.io/dns
 script:
 - make build
 - make test
-- bash test/e2e/sidecar/e2e.sh
 - make containers
+- sudo -v
+- cd ${GOPATH}/k8s.io/dns && GOPATH=$GOPATH:`pwd`/vendor bin/amd64/ginkgo test/e2e
```

Comment:
There is no need to cd into ${GOPATH}/src/k8s.io/dns You are already in (ln -s on install stage)